File: POO/Java/Herança/Exercicio 3/Funcionario.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.projetojava;

/**
 *
 * @author Aluno
 */
public abstract class Funcionario extends Fisica{
    protected String cpf;
    protected String rg;
    protected String matricula;
    protected Setor setor;
    protected float salario;

    public Funcionario(String cpf, String rg, String matricula, Setor setor, float salario, Sexo sexo, EstadoCivil estadoCivil, String dataNascimento, int id, String nome, String telefone, String email, Endereco endereco) {
        super(sexo, estadoCivil, dataNascimento, id, nome, telefone, email, endereco);
        this.cpf = cpf;
        this.rg = rg;
        this.matricula = matricula;
        this.setor = setor;
        this.salario = salario;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public String getRg() {
        return rg;
    }

    public void setRg(String rg) {
        this.rg = rg;
    }

    public String getMatricula() {
        return matricula;
    }

    public void setMatricula(String matricula) {
        this.matricula = matricula;
    }

    public Setor getSetor() {
        return setor;
    }

    public void setSetor(Setor setor) {
        this.setor = setor;
    }

    public float getSalario() {
        return salario;
    }

    public void setSalario(float salario) {
        this.salario = salario;
    }

    @Override
    public String toString() {
        return "\n" + 
                super.toString() +
                "\nCpf: " + cpf + 
                "\nRg: " + rg + 
                "\nMatricula: " + matricula + 
                "\nSetor: " + setor + 
                "\nSálario=" + salario;
    }
    
    
    
}

File: estruturadeprojetos/projeto_de_exercicio/models/funcionario.py
from models.enum.setor import Setor
from models.enum.sexo import Sexo
from models.enum.unidade_federativa import unidadeFederativa
from models.endereco import Endereco

class Funcionario:
    def __init__(self,id: int,nome: str,cpf: str,rg: str,matricula: str,dataNascimento: str,sexo: Sexo,setor: Setor,salario: float,telefone: str,email: str,endereco: Endereco) -> None:
        self.id = id
        self.nome = nome
        self.cpf = cpf
        self.rg = rg
        self.matricula = matricula
        self.dataNascimento = dataNascimento
        self.sexo = sexo
        self.setor = setor
        self.salario = salario
        self.telefone = telefone
        self.email = email
        self.endereco = endereco

    def __str__(self) -> str:
        return(f"\nId: {self.id}"
               f"\nNome: {self.nome}"
               f"\nCpf: {self.cpf}"
               f"\nRg: {self.rg}"
               f"\nMatricula: {self.matricula}"
               f"\nData de Nascimento: {self.dataNascimento}"
               f"\nSetor: {self.setor.value}"
               f"\nSálario: R$ {self.salario}"
               f"\nTelefone: {self.telefone}"
               f"\nE-mail: {self.email}"
               f"\nEndereço: {self.endereco}"
               )